src/cryptonote_core/blockchain.cpp

```diff
@@ -1142,6 +1142,12 @@ bool Blockchain::create_block_template(block& b, const account_public_address& m
   b.prev_id = get_tail_id();
   b.timestamp = time(NULL);
 
+  uint64_t median_ts;
+  if (!check_block_timestamp(b, median_ts))
+  {
+    b.timestamp = std::max(median_ts, m_db->get_top_block_timestamp());
```

Comment:
That seems dangerous, no ? It means the miner of that block gets an inordinate say over what subsequent miners will timestamp with.

Reply:
Would you prefer the following? `b.timestamp = std::max(median_ts, m_db->get_top_block_timestamp() - CRYPTONOTE_BLOCK_FUTURE_TIME_LIMIT);`

Reply:
I'd rather use b.timestamp = median_ts; as this is the smallest "sanity" change which makes the block pass consensus rules.

Reply:
That's reasonable - I've changed it back to `b.timestamp = median_ts;`